### PR TITLE
Add docs for template placeholders

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
Babel PR: https://github.com/babel/babel/pull/9648

### CODE REVIEW

File: docs/template.md [hunk 1]
```diff
@@ -20,22 +20,37 @@ import generate from "@babel/generator";
 import * as t from "@babel/types";
 
 const buildRequire = template(`
-  var IMPORT_NAME = require(SOURCE);
+  var %%importName%% = require(%%source%%);
```
Comment (existentialism) [inline docs/template.md]: We should probably note that using this syntax requires `syntacticPlaceholders`?

I think we should keep the original docs and add a "check out this new hotness via this option" section right after it?

Comment (nicolo-ribaudo) [reply docs/template.md]: No, it doesn't. If you are using `%%foo%%`, that option is automatically enabled.

Comment (existentialism) [reply docs/template.md]: Sounds good.

> You can use either new or old placeholders by default, you just can't mix them.

Seems necessary to mention though?

File: docs/template.md [hunk 2]
```diff
@@ -27,15 +47,15 @@ const ast = buildRequire({
...
 ```
 
+Output:
+
 ```js
 const myModule = require("my-module");
 ```
 
-### `.ast`
+### `.ast`!
```
Comment (nicolo-ribaudo) [inline docs/template.md]:
```suggestion
### `.ast`
```

**Pull request merged**